packages/astro/src/core/config/config.ts

```diff
@@ -172,11 +190,19 @@ async function loadConfig(
 	if (!configPath) return {};
 
 	// Create a vite server to load the config
-	return await loadConfigWithVite({
-		root,
-		configPath,
-		fs: fsMod,
-	});
+	try {
+		return await loadConfigWithVite({
+			root,
+			configPath,
+			fs: fsMod,
+		});
+	} catch (e) {
+		const configPathText = configFile ? colors.bold(configFile) : 'your Astro config';
+		// Config errors should bypass log level as it breaks startup
```

Comment:
About the comment: Is it because `--silent` can potentially not log these errors?

Reply:
Yeah `--silent` would otherwise not log that the config failed to load. Try to pass the `logging` down here is a bit tricky, especially the fact that an Astro config `logLevel` can also contribute to how `logging` works.

Reply:
Actually there's no `logLevel` config in the Astro config, got it mixed up. I think it's still probably fine always logging it here as the other code in this file also does that, but I'll revisit this again if it's not ideal.